docs/001_develop/03_client-capabilities/022_framework-integration/002_integration-angular.mdx

```diff
@@ -0,0 +1,279 @@
+
+---
+
+### src/pbc
+This folder contains components that are responsible for enabling the insertion of slots within the application.
+
+These slots act as placeholders where content, provided by the registered Project Building Components ([PBCs](../../../../server/packaged-business-capabilities/pbc-intro/)), is rendered dynamically. The [PBCs](../../../../server/packaged-business-capabilities/pbc-intro/) are registered in the application through **main.ts**, ensuring that specific components can be inserted into the designated slots at runtime.
+
+---
+
+### src/app/share
+The **share** folder holds shared resources and components that are used across the entire application.
+
+Key files are:
+
+- **genesis-components.ts**. This registers Genesis framework components, including forms, layouts, and charts.
+- **foundation-login.ts**. This configures the foundation-login micro front-end component that handles authorisation and integrates it with the routing system.
+
+---
+
+### src/app/pages
+This folder contains the main pages of the application. Each page represents a different route or view, such as the **auth-login.component.ts**, which handles authentication-related flows.
+
+---
+
+### src/app/components
+This folder contains reusable UI components that are utilized throughout the app. Components in this folder are not tied to specific pages but are used as building blocks across multiple sections of the application.
+
+## Routing
+
+In Angular, routing is essential for creating single-page applications with navigation capabilities.
+
+The routing configuration in **app/service/route.service.ts** manages and provides routes throughout the application. This file defines a `RouteService` service that combines routes from the main application and additional routes from PBC (Pluggable Business Components).
+
+The `pbcRoutes` are generated dynamically by mapping over the routes provided by the PBC, extracting essential properties, and wrapping them in a `PBCContainer` component. These routes are then combined with the main application's routes into a single `allRoutes` array.
+
+The `RoutesProvider` uses Angular's dependency injection to make these routes available to the rest of the application by providing them through the `RoutesService`.
+
+Additionally, the `RoutesService` is provided to access the routes within Angular components. This ensures seamless integration and accessibility of both main application routes and PBC routes throughout the application.
+
+## Setting attributes
+
+Attributes are part of the HTML markup of an element and are reflected in the dom.
+
+Attributes can be either hard coded or set using Angular's property event bindings. Note, any automatically maps hyphenated attribute names to camelCase property names so a property called `max-rows` becomes `maxRows`.

```

Comment:
"Note, any automatically maps hyphenated attribute names to camelCase property" this sentence doesn't sound correct